common/common/src/main/java/io/helidon/build/common/FileUtils.java

```diff
@@ -742,23 +743,42 @@ public static FileSystem newZipFileSystem(Path zip) {
      * @return zip file
      */
     public static Path zip(Path zip, Path directory) {
+        return zip(zip, directory, path -> {});
+    }
+
+    /**
+     * Zip a directory.
+     *
+     * @param zip           target file
+     * @param directory     source directory
+     * @param fileConsumer  zipped file consumer
+     * @return zip file
+     */
+    public static Path zip(Path zip, Path directory, Consumer<Path> fileConsumer) {
         ensureDirectory(zip.getParent());
         try (FileSystem fs = newZipFileSystem(zip)) {
+            boolean posix = isPosix(directory);
             try (Stream<Path> entries = Files.walk(directory)) {
                 entries.sorted(Comparator.reverseOrder())
                        .filter(p -> Files.isRegularFile(p) && !p.equals(zip))
-                       .forEach(p -> {
+                       .map(p -> {
                            try {
                                Path target = fs.getPath(directory.relativize(p).toString());
                                Path parent = target.getParent();
                                if (parent != null) {
                                    Files.createDirectories(parent);
                                }
                                Files.copy(p, target, REPLACE_EXISTING);
+                               if (posix) {
```

Comment:
This is not what I described in the issue. 

IIRC the permissions are already preserved when running on posix, at least that's what I remember seeing when I tested.

On Windows `Files.getPosixFilePermissions(p)` won't work, so the idea of the `Consumer` is to allow setting the permissions manually.

We convert the existing `zip` method to take a `Consumer` and overload it to preserve the old signature.
I.e.

```java
public static Path zip(Path zip, Path directory, Consumer<Path> fileConsumer) {
    return zip(zip, directory,  p -> {});
}
```

Then that allows `Windows` code to do something like this:
```java
zip(zip, directory, 
    p -> unchecked(() -> Files.setPosixFilePermissions(null, Set.of(OTHERS_EXECUTE))));
```